google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/scaladsl/BigQuery.scala

```diff
@@ -0,0 +1,513 @@
+      tableDataListUnmarshaller: Unmarshaller[Json, TableDataListResponse[Out]]
+  ): Source[Out, Future[TableDataListResponse[Out]]] =
+    source { settings =>
+      val uri = BigQueryEndpoints.tableData(settings.projectId, datasetId, tableId)
+      val query = Query.Empty :+?
+        "startIndex" -> startIndex :+?
+        "maxResults" -> maxResults :+?
+        "selectedFields" -> (if (selectedFields.isEmpty) None else Some(selectedFields.mkString(",")))
+      paginatedRequest[TableDataListResponse[Out], Json](HttpRequest(GET, uri.withQuery(query)))
+    }.wireTapMat(Sink.head)(Keep.right).mapConcat(_.rows.fold[List[Out]](Nil)(_.toList))
+
+  def insertAll[In](
+      datasetId: String,
+      tableId: String,
+      retryPolicy: InsertAllRetryPolicy,
+      templateSuffix: Option[String] = None
+  )(implicit marshaller: ToEntityMarshaller[TableDataInsertAllRequest[In]]): Sink[Seq[In], NotUsed] = {
+    val requests = Flow[Seq[In]].statefulMapConcat { () =>
+      val randomGen = new SplittableRandom
+
+      xs => {
+        val rows = xs.map { x =>
+          val insertId =
+            if (retryPolicy.deduplicate)
+              Some(randomUUID(randomGen).toString)
+            else
+              None
+          TableDataJsonProtocol.Row(insertId, x)
+        }
+
+        TableDataInsertAllRequest(None, None, templateSuffix, rows) :: Nil
+      }
+    }
+
+    val errorSink = Sink.foreach[TableDataInsertAllResponse] { response =>
+      response.insertErrors
+        .flatMap(_.headOption)
+        .flatMap(_.errors)
+        .flatMap(_.headOption)
+        .foreach(error => throw BigQueryException(error))
+    }
+
+    requests.via(insertAll(tableId, datasetId, retryPolicy.retry)).to(errorSink)
+  }
+
+  def insertAll[In](datasetId: String, tableId: String, retryFailedRequests: Boolean)(
+      implicit marshaller: ToEntityMarshaller[TableDataInsertAllRequest[In]]
+  ): Flow[TableDataInsertAllRequest[In], TableDataInsertAllResponse, NotUsed] =
+    Flow
+      .fromMaterializer { (mat, attr) =>
+        import BigQueryException._
+        import SprayJsonSupport._
+        import mat.executionContext
+        implicit val system = mat.system
+        implicit val settings = BigQueryAttributes.resolveSettings(attr, mat)
+
+        val uri = BigQueryEndpoints.tableDataInsertAll(settings.projectId, datasetId, tableId)
+        val request = HttpRequest(POST, uri)
+
+        val http = BigQueryHttp()
+        val requestWithOAuth =
+          if (retryFailedRequests)
+            http.retryRequestWithOAuth(_)
+          else
+            http.singleRequestWithOAuthOrFail(_)
+
+        Flow[TableDataInsertAllRequest[In]]
+          .mapAsync(1)(Marshal(_).to[RequestEntity])
+          .map(request.withEntity)
+          .mapAsync(1)(requestWithOAuth)
+          .mapAsync(1)(Unmarshal(_).to[TableDataInsertAllResponse])
+      }
+      .mapMaterializedValue(_ => NotUsed)
+
+  def insertAllAsync[In](datasetId: String,
```

Comment:
Thoughts on method name? It is similar in functionality to `insertAll` (adding rows to an existing table). However `insertAll` implements [(near) real-time streaming inserts](https://cloud.google.com/bigquery/streaming-data-into-bigquery), whereas `insertAllAsync` creates load jobs that ["execute asynchronously and can be polled for their status."](https://cloud.google.com/bigquery/docs/jobs-overview)